### PR TITLE
Feature/3114 hero tweaks

### PR DESCRIPTION
**Pull request by emclaug2**
#### Changes proposed in this Pull Request:
-  Wrapped all nested elements in Hero with `styled`, but Icon classes are not working for size.

### CODE REVIEW

File: components/src/core/Hero/Hero.tsx [hunk 1]
```diff
@@ -73,27 +74,29 @@ const Root = styled(Box, { name: 'hero', slot: 'root' })<
...
-        width: '100%',
-        textAlign: 'center',
-        overflow: 'hidden',
-        textOverflow: 'ellipsis',
-        whiteSpace: 'nowrap',
-    },
+    })
+);
+
+const Values = styled(Box, { name: 'hero', slot: 'values' })<Pick<HeroProps, undefined>>(({ theme }) => ({
```
Comment (daileytj) [inline components/src/core/Hero/Hero.tsx:80]: You should be able to remove the Hero props <pick> type definition if you aren't passing in any props

File: components/src/core/Hero/Hero.tsx [hunk 2]
```diff
@@ -117,20 +120,24 @@ const HeroRender: React.ForwardRefRenderFunction<unknown, HeroProps> = (props: H
...
+            </Icon>
+            <Values className={cx(defaultClasses.values, classes.values)}>
                 {!props.children && ChannelValueProps?.value && (
                     <ChannelValue fontSize={ChannelValueProps?.fontSize} {...ChannelValueProps} />
                 )}
                 {props.children}
-            </span>
-            <Typography variant={'body1'} color={'inherit'} className={cx(defaultClasses.label, classes.label)}>
+            </Values>
+            <Label variant={'body1'} color={'inherit'} className={cx(defaultClasses.label, classes.label)}>
```
Comment (daileytj) [inline components/src/core/Hero/Hero.tsx:138]: The `Box` component is a `div` by default. Label and Values should be `span`. 
```suggestion
            <Label variant={'body1'}  component={'span'} color={'inherit'} className={cx(defaultClasses.label, classes.label)}>
```

File: components/src/core/Hero/Hero.tsx [hunk 3]
```diff
@@ -117,20 +120,24 @@ const HeroRender: React.ForwardRefRenderFunction<unknown, HeroProps> = (props: H
...
-            <span className={cx(defaultClasses.icon, classes.icon)}>{icon}</span>
-            <span className={cx(defaultClasses.values, classes.values)}>
+            <Icon
+                className={cx(defaultClasses.icon, classes.icon)}
+                iconSize={iconSize}
+                iconBackgroundColor={iconBackgroundColor}
+            >
+                {icon}
+            </Icon>
+            <Values className={cx(defaultClasses.values, classes.values)}>
```
Comment (daileytj) [inline components/src/core/Hero/Hero.tsx:132]: The `Box` component is a `div` by default. Label and Values should be `span`.

```suggestion
            <Values  component={'span'} className={cx(defaultClasses.values, classes.values)}>
```

File: components/src/core/Hero/Hero.tsx [hunk 4]
```diff
@@ -117,20 +120,24 @@ const HeroRender: React.ForwardRefRenderFunction<unknown, HeroProps> = (props: H
...
+            </Icon>
+            <Values  component={'span'} className={cx(defaultClasses.values, classes.values)}>
                 {!props.children && ChannelValueProps?.value && (
                     <ChannelValue fontSize={ChannelValueProps?.fontSize} {...ChannelValueProps} />
                 )}
                 {props.children}
-            </span>
-            <Typography variant={'body1'} color={'inherit'} className={cx(defaultClasses.label, classes.label)}>
+            </Values>
+            <Label variant={'body1'}  component={'span'} color={'inherit'} className={cx(defaultClasses.label, classes.label)}>
```
Comment (bkarambe) [inline components/src/core/Hero/Hero.tsx:138]: As we are passing component={'span'} to Label component, we need to give type to the styled Label component. Typography does not accept component prop, so we need to give <TypographyProps & BoxProps>.

**Review by bkarambe — CHANGES_REQUESTED** [review]
Please run prettier.

File: components/src/core/Hero/Hero.tsx (hunk 4 above)
Comment (emclaug2) [reply components/src/core/Hero/Hero.tsx]: I've removed the `component={span}` since Label is a Typography & not a Box.

**Review by bkarambe — APPROVED** [review]

**Pull request merged**